src/TestSuite/Fixture/TestFixture.php

```diff
@@ -195,11 +201,23 @@ protected function _getRecords(): array
         $values = [];
         $types = [];
         $columns = $this->_schema->columns();
-        foreach ($this->records as $record) {
-            $fields = array_merge($fields, array_intersect(array_keys($record), $columns));
+        foreach ($this->records as $index => $record) {
+            $recordFields = array_keys($record);
+            if ($this->strictFields) {
+                $invalidFields = array_values(array_filter($recordFields, fn ($f) => !in_array($f, $columns, true)));
+                if ($invalidFields !== []) {
+                    throw new CakeException(
+                        "Record #{$index} in fixture has invalid fields: " . json_encode($invalidFields)
```

Comment:
```suggestion
                        "Record #{$index} in fixture has additional fields that do not exist in the schema. Remove the following fields: " . json_encode($invalidFields)
```

We can give a bit more guidance on how to resolve this problem.